tests/unit/backend/chalice/api_server/test_revisions.py

```diff
@@ -116,3 +120,202 @@ def test__publish_revision__201(self):
             self.assertIn(link, actual_body["links"])
         for keys in expected_body.keys():
             self.assertEqual(expected_body[keys], actual_body[keys])
+
+    def test__revision_deleted_with_published_datasets(self):
+        """The published dataset artifacts should be intact after deleting a collection revision"""
+        # Generate public collection
```

Comment:
Should we start pulling this out as a testing fixture/extension of the generate collection options?